Test/PacketType/Ieee80211/PerPacketInformationTest.cs

```diff
@@ -247,6 +247,7 @@ public void RemoveField()
                 dev.Close ();
                 
                 PpiPacket p = Packet.ParsePacket (rawCapture.LinkLayerType, rawCapture.Data) as PpiPacket;
+                Assert.IsTrue(p?.PayloadPacket is MacFrame);
```

Comment:
These three tests otherwise fail, this is because after removing a field, the original packet cannot parse the Mac frame anymore as a field has been added/removed. In `UpdateCalculatedValues` the header size gets then modified and then it doesn't think there's a Mac frame anymore. In this instance, I think it's better to populate the payload packet before removing/adding a field. I'll commit this change now.